### PR TITLE
Feat ajax

### PR DESCRIPTION
**Pull request by mixcas**

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -110,6 +111,116 @@ var Menu = {
...
+    _this.$ajaxyLinks.click(function(event) {
+      event.preventDefault();
+
+      var url = event.currentTarget.href;
+
+      _this.ajaxLoad(url);
+
+    });
+
+    $(document).ready( function() {
```
Comment (varffs) [inline js/main.js:134]: is this just binding in another place?

File: js/main.js [hunk 2]
```diff
@@ -110,6 +111,116 @@ var Menu = {
...
+
+  ajaxAfter: function() {
+    var _this = this;
+
+    $('body').removeClass('loading');
+
+    _this.reset();
+
+    // Resets from other parts of the website
+    Twitter.init();
```
Comment (varffs) [inline js/main.js:192]: should this be conditional? for things like body class etc? `if ($('body').hasClass('home'))`

File: js/main.js (hunk 1 above)
Comment (mixcas) [inline js/main.js:134]: what do u mean?

File: js/main.js (hunk 2 above)
Comment (mixcas) [inline js/main.js:192]: I think that logic should go inside `Twitter`

File: js/main.js (hunk 1 above)
Comment (varffs) [inline js/main.js:134]: there is already a doc ready. Does this just bind another function to that event?

Comment (mixcas) [inline js/main.js:134]: Yeah

File: js/main.js [hunk 3]
```diff
@@ -110,6 +111,116 @@ var Menu = {
...
+
+  },
+
+  ajaxErrorHandler: function(jqXHR, textStatus) {
+    alert(textStatus);
+    console.log(jqXHR);
+  },
```
Comment (varffs) [inline js/main.js:199]: does `complete` run after an error or no? If so not much left behind in markup on an error

File: js/main.js [hunk 4]
```diff
@@ -124,20 +235,21 @@ jQuery(document).ready(function () {
   Twitter.init();
   Menu.init();
 
+  Ajaxy.init('#drawer-categorias-list a, #footer-categorias-list a');
```
Comment (varffs) [inline js/main.js]: this function doesnt have args anymore

File: js/main.js (hunk 2 above)
Comment (varffs) [inline js/main.js:192]: hmm yh fair enough

File: js/main.js (hunk 4 above)
Comment (mixcas) [inline js/main.js]: fixing it

File: js/main.js (hunk 3 above)
Comment (mixcas) [inline js/main.js:199]: yes

File: js/main.js [hunk 5]
```diff
@@ -110,6 +111,116 @@ var Menu = {
...
+    $('body, html').animate({
+      scrollTop: 0,
+    }, fastAnimationSpeed);
+  },
+
+  ajaxAfter: function() {
+    var _this = this;
+
+    $('body').removeClass('loading');
+
```
Comment (varffs) [inline js/main.js:188]: not going to block merge for this but think the Menu function to close the drawers other than radio should trigger here

**Comment by varffs** [comment]
need additional functionality for search as form submit isnt a bound `<a>` link. but not going to block merge for that

**Comment by mixcas** [comment]
@varffs true, rise an issue

**Comment by mixcas** [comment]
@varffs cuz its bad UX

**Comment by varffs** [comment]
ok raising 2 issues for things and going to merge this as is

**Pull request merged**